cmd/project.go

```diff
@@ -145,6 +146,21 @@ var projectDeleteCmd = &cobra.Command{
 			checkError(err, "")
 		}
 		fmt.Printf("Deleted project : %v\n", projectName)
+
+		// Wait for the delete operation to reflect in the projects list
+		time.Sleep(9 * time.Second)
+
+		// Get Current Project
+		currProject := project.GetCurrent(client)
+
+		// Check if List returns empty, if so, the currProject is showing old currentProject
```

Comment:
No it is the same even in case of oc client..
Verified using `oc new-project prj4;oc delete project prj4;sleep 6; oc projects;oc project`
@cdrage

Reply:
6 seconds isn't a magic number, we should be determining if the project has been deleted through either a loop or simply waiting for `project.Delete` to finish. We should never have `time.Sleep` in the code..

Reply:
But post the delete of the only remaining project, openshift doesn't seem to unset or set the current/active project as can be seen in:

```
[redhat@localhost odo]$ oc project
error: you do not have rights to view project "myproject".
```

So, we have 2 choices:
1. Set ONLY in our client, the namespace as "" and handle it separately in our code(I feel this can be buggy bcoz occlient namespace on my machine is different from that on some other machine even if we connect to same cluster as it is only local state and not really across machines as its very much within odo and specific to a installation)
2. Current implementation: get list of projects, if length 0 --> No projects available else display currentProject as active project